modules/tinymce/src/core/main/ts/api/file/BlobCache.ts

```diff
@@ -90,30 +91,25 @@ export const BlobCache = function (): BlobCache {
     };
   };
 
-  const add = function (blobInfo: BlobInfo) {
+  const add = (blobInfo: BlobInfo) => {
     if (!get(blobInfo.id())) {
       cache.push(blobInfo);
     }
   };
 
-  const get = function (id: string): BlobInfo {
-    return findFirst(function (cachedBlobInfo) {
-      return cachedBlobInfo.id() === id;
-    });
-  };
+  const findFirst = (predicate: (blobInfo: BlobInfo) => boolean) => Arr.find(cache, predicate).getOrUndefined();
 
-  const findFirst = function (predicate: (blobInfo: BlobInfo) => boolean) {
-    return Arr.filter(cache, predicate)[0];
-  };
+  const get = (id: string) =>
+    findFirst((cachedBlobInfo) => cachedBlobInfo.id() === id);
 
-  const getByUri = function (blobUri: string): BlobInfo {
-    return findFirst(function (blobInfo) {
-      return blobInfo.blobUri() === blobUri;
-    });
-  };
+  const getByUri = (blobUri: string) =>
+    findFirst((blobInfo) => blobInfo.blobUri() === blobUri);
+
+  const getByData = (base64: string, type: string) =>
```

Comment:
Funny story, the original `BlobCache` has _always_ had a `lookupByData` method. Not sure why this one didn't.

Reply:
Haha, so it did. I did actually originally call this `lookupByData` as well, but then changed it to `getByData` for naming consistency :shrug: